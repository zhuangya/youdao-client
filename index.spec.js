'use strict';

jest.mock('superagent');

const Youdao = require('./');

const y = new Youdao({
  key: 'key',
  secret: 'secret'
});

it('should sign request', () => {
  expect(y.sign('hello', 123)).toMatchSnapshot();
});

it('should build the correct query', () => {
  expect(y.buildQuery('world', { salt: 'salt' })).toMatchSnapshot();
});

it('should set from & to properly when translate', async () => {
  const res = await y.translate('Amigo', { from: 'es', to: 'en', salt: 'fixed' });
  expect(res).toMatchSnapshot();
});

it('should throw err', async () => {
  const broken = new Youdao({
    key: 'not-found',
    secret: 'shhhh',
    base: 'https://out.of.order/api'
  });

  try {
    await broken.translate('voodoo')
  } catch (e) {
    expect(e).toMatchSnapshot();
  }
});
