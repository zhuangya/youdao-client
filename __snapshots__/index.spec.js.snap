// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should build the correct query 1`] = `
Object {
  "appKey": "key",
  "from": "auto",
  "q": "world",
  "salt": "salt",
  "sign": "adb3038fee34c0a75c93e53f15a8c8d8",
  "to": "auto",
}
`;

exports[`should set from & to properly when translate 1`] = `"https://openapi.youdao.com/api?from=es&to=en&q=Amigo&appKey=key&salt=fixed&sign=62fbddbc961c6edc89dfe9643762fe4a"`;

exports[`should sign request 1`] = `"fe764d51dd60550574e76c600bdfdbd2"`;

exports[`should throw err 1`] = `"500"`;
